clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12216,6 +12216,9 @@ def err_std_source_location_impl_not_found : Error<
 def err_std_source_location_impl_malformed : Error<
   "'std::source_location::__impl' must be standard-layout and have only two 'const char *' fields '_M_file_name' and '_M_function_name', and two integral fields '_M_line' and '_M_column'">;
 
+def err_std_initializer_list_malformed : Error<
+  "%0 layout not recognized. Must be a struct with two fields, a 'const E *' and either another 'const E *' or a 'std::size_t'">;
```

Comment:
```suggestion
  "%0 layout not recognized. Must be a non-polymorphic class type with no bases and two fields: a 'const E *' and either another 'const E *' or a 'std::size_t'">;
```
I do realise that we’re cramming quite a bit of information into this diagnostic here, but so does `err_std_source_location_impl_malformed`, and this is not a diagnostic anyone except for library implementers should ever see, so this is probably fine imo.